### PR TITLE
[INCLUDE][SHELL32_APITEST] Add SHLimitInputEdit testcase

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-11701](https://jira.reactos.org/browse/CORE-11701)
- Add `IItemNameLimits` interface into `shobjidl.idl`.
- Add a testcase for `shell32!SHLimitInputEdit` function.

### CODE REVIEW

**Comment by katahiromz** [comment]
WinXP:
shlimit-winxp
Successful.

Win2k3:
shlimit-win2k3
Successful.

Win10:
shlimit-win10
Successful.

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+#include <shobjidl.h>
+#include <shlwapi.h>
+
+class CShellFolder
+    : public IShellFolder
+    , public IItemNameLimits
+{
+public:
+    CShellFolder(INT iMaxNameLen = 0, BOOL bDisabled = FALSE)
+        : m_nRefCount(1)
```
Comment (learn-more) [inline modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: Your reference counting is off, it should be initialized on 0.

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    trace("QueryInterface\n");
+
+    if (!ppvObject)
+        return E_POINTER;
+
+    if (IsEqualIID(riid, IID_IItemNameLimits))
+    {
+        trace("IID_IItemNameLimits\n");
+        *ppvObject = static_cast<IItemNameLimits *>(this);
+        return S_OK;
```
Comment (learn-more) [inline modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: A succesful match in `QueryInterface` should call `AddRef`

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        hr = SHLimitInputEdit(hEdt1, NULL);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        hr = 0xDEAD;
+    }
+    _SEH2_END;
+    ok_int(hr, 0xDEAD);
+
+    CComPtr<IShellFolder> psf(new CShellFolder(123, FALSE));
```
Comment (learn-more) [inline modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: However, none of that reference counting really matters for the test case?

You can just create the object here like this:
`CShellFolder psf(123, FALSE);`

And just update the `AddRef` to always return 2, `Release` to always return 1.

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp (hunk 2 above)
Comment (katahiromz) [reply modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: OK.

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp (hunk 3 above)
Comment (katahiromz) [reply modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: OK.

File: modules/rostests/apitests/shell32/SHLimitInputEdit.cpp (hunk 1 above)
Comment (katahiromz) [reply modules/rostests/apitests/shell32/SHLimitInputEdit.cpp]: The reference count is important for this case. We have to investigate whether `AddRef` and `Release` are called.

**Pull request merged**